front/app/modules/commercial/content_builder/admin/components/CraftComponents/Iframe/index.tsx

```diff
@@ -0,0 +1,149 @@
+    height: node.data.props.height,
+    id: node.id,
+    hasError: node.data.props.hasError,
+  }));
+
+  return (
+    <Box flexWrap="wrap" display="flex" gap="16px" marginBottom="20px">
+      <Box flex="0 0 100%">
+        <Input
+          label={
+            <span>
+              {formatMessage(messages.iframeUrlLabel)}{' '}
+              <IconTooltip
+                icon="info3"
+                content={formatMessage(messages.iframeUrlLabelTooltip)}
+              />
+            </span>
+          }
+          placeholder={formatMessage(messages.iframeUrlPlaceholder)}
+          type="text"
+          value={url}
+          onChange={(value) => {
+            setProp((props) => (props.url = value));
+            setProp((props) => (props.hasError = !validateUrl(value)));
+            eventEmitter.emit('contentBuilderError', {
+              [id]: !validateUrl(value),
+            });
+          }}
+        />
+        {hasError && <Error marginTop="4px" text={'Error Message'} />}
+      </Box>
+      <Box flex="0 0 100%">
+        <Input
+          label={
+            <span>
+              {formatMessage(messages.iframeHeightLabel)}{' '}
+              <IconTooltip
+                icon="info3"
+                content={formatMessage(messages.iframeHeightLabelTooltip)}
+              />
+            </span>
+          }
+          placeholder={formatMessage(messages.iframeHeightPlaceholder)}
+          type="text"
+          value={height}
+          onChange={(value) => {
+            setProp((props) => (props.height = value));
+          }}
+        />
+      </Box>
+    </Box>
+  );
+});
+
+Iframe.craft = {
+  props: {
+    url: '',
+    height: '',
+  },
+  related: {
+    settings: IframeSettings,
+  },
+};
+
+/*
+ * Function to validate embed URL against white list
+ * Returns: boolean value whether URL input string is valid or not
+ */
+const validateUrl = (url: string) => {
+  const urlWhiteList = [
+    /https:\/\/.*\.typeform\.com\/to\/.*/,
+    /https:\/\/widget\.surveymonkey\.com\/collect\/website\/js\/.*\.js/,
+    /https:\/\/docs.google.com\/forms\/d\/e\/.*\/viewform\?embedded=true/,
+    /https:\/\/surveys.enalyzer.com\/?\?pid=.*/,
+    /https:\/\/www\.survey-xact\.dk\/LinkCollector\?key=.*/,
+    /https:\/\/.*\.qualtrics\.com\/jfe\/form\/.*/,
+    /https:\/\/www\.smartsurvey\.co\.uk\/.*/,
+    /https:\/\/.*\.(microsoft|office)\.com\//,
+    /www\.eventbrite\.com\/static\/widgets\/*/,
+    /https:\/\/.*arcgis\.com/,
+    /https:\/\/public\.tableau\.com.*/,
+    /datastudio\.google\.com\/embed*/,
+    /https:\/\/app\.powerbi\.com\//,
+    /static\.ctctcdn\.com\/js*/,
+    /http:\/\/.*instagram\.com/,
+    /https:\/\/.*platform\.twitter\.com/,
+    /https:\/\/name\.konveio\.com/,
+    /https:\/\/.*youtu(be|.be)*/,
+    /https:\/\/.*vimeo.com/,
+    /https:\/\/.*dailymotion\.com\/embed\/video/,
+    /https:\/\/.*media\.videotool\.dk\/.*vn=/,
+    /https:\/\/.*dreambroker.com\/channel\/.*\/iframe/,
+  ];
```

Comment:
For the survey validation on the front end, we require https. Should I assume the same for the rest of the whitelisted url sources?